cmd/integrationArtifactTriggerIntegrationTest_test.go

```diff
@@ -0,0 +1,218 @@
+package cmd
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/SAP/jenkins-library/pkg/mock"
+	"github.com/stretchr/testify/assert"
+)
+
+type integrationArtifactTriggerIntegrationTestMockUtils struct {
+	*mock.ExecMockRunner
+	*mock.FilesMock
+}
+
+func newIntegrationArtifactTriggerIntegrationTestTestsUtils() integrationArtifactTriggerIntegrationTestMockUtils {
+	utils := integrationArtifactTriggerIntegrationTestMockUtils{
+		ExecMockRunner: &mock.ExecMockRunner{},
+		FilesMock:      &mock.FilesMock{},
+	}
+	return utils
+}
+
+func TestRunIntegrationArtifactTriggerIntegrationTest(t *testing.T) {
+	//t.Parallel()
+	//t.Run("happy path", func(t *testing.T) {
+	//	t.Parallel()
+	//	// init
+	//	config := integrationArtifactTriggerIntegrationTestOptions{
+	//		Host:                  "https://demo",
+	//		OAuthTokenProviderURL: "https://demo/oauth/token",
+	//		Username:              "demouser",
+	//		Password:              "******",
+	//		IntegrationFlowID:     "CPI_IFlow_Call_using_Cert",
+	//		Platform:              "cf",
+	//	}
+	//
+	//	utils := newintegrationArtifactTriggerIntegrationTestTestsUtils()
+	//	utils.AddFile("file.txt", []byte("dummy content"))
+	//	httpClient := httpMockCpis{CPIFunction: "IntegrationArtifactGetServiceEndpoint", ResponseBody: ``, TestType: "PositiveAndGetetIntegrationArtifactGetServiceResBody"}
+	//
+	//	// test
+	//	err := runintegrationArtifactTriggerIntegrationTest(&config, nil, utils, &httpClient)
+	//
+	//	// assert
+	//	assert.NoError(t, err)
+	//})
+
+	t.Run("MessageBodyPath good but no ContentType (ERROR) callIFlowURL", func(t *testing.T) {
+		//init
+		iFlowServiceKey := `{
+			"oauth": {
+				"url": "https://demo",
+				"clientid": "demouser",
+				"clientsecret": "******",
+				"tokenurl": "https://demo/oauth/token"
+			}
+		}`
+		config := integrationArtifactTriggerIntegrationTestOptions{
+			IFlowServiceKey:   iFlowServiceKey,
+			IntegrationFlowID: "CPI_IFlow_Call_using_Cert",
+			Platform:          "cf",
+			MessageBodyPath:   "/file.txt",
+			ContentType:       "",
+		}
+
+		utils := newIntegrationArtifactTriggerIntegrationTestTestsUtils()
+		utils.AddFile("file.txt", []byte("dummycontent"))
+		httpClient := httpMockCpis{CPIFunction: "IntegrationArtifactGetServiceEndpoint", ResponseBody: ``, TestType: "PositiveAndGetetIntegrationArtifactGetServiceResBody"}
```

Comment:
Here the wrong mocking CPI function is called. We should add `IntegrationArtifactTriggerIntegrationTest` to the mock.

Reply:
Would there be any more steps to that than just changing the name in the `httpClient` of all the tests?